test/e2e/tests/block_test.go

```diff
@@ -55,14 +57,41 @@ func TestBlock_Header(t *testing.T) {
 	})
 }
 
+// Tests that the node configured to prune are actually pruning.
+func TestBlock_Pruning(t *testing.T) {
+	t.Helper()
+	testNode(t, func(t *testing.T, node e2e.Node) {
+		t.Helper()
+		// We do not run this test on seed nodes or nodes with data
+		// companion-related pruning enabled or nodes not pruning.
+		if node.Mode == e2e.ModeSeed || node.EnableCompanionPruning || node.RetainBlocks == 0 {
```

Comment:
Should this also exclude light nodes ?

```suggestion
		if node.Mode == e2e.ModeSeed || node.Mode == e2e.ModeLight || node.EnableCompanionPruning || node.RetainBlocks == 0 {
```

or use the `node.Stateless()` to exclude `light` and `seed` nodes
```suggestion
		if node.Mode == node.Stateless() || node.EnableCompanionPruning || node.RetainBlocks == 0 {
```

Reply:
Thing is, I don't fully understand why we were excluding seed nodes in the first place, so I just left it there. On the other hand, light nodes seem to be passing the test anyway, so I'd leave them in.
Maybe we should try removing the check for seed, and see if the test still passes. WDYT?

Reply:
I tried removing the seed on one test case, and followed your advice on the other, and used `Stateless()`. I tried the CI manifeests several times and they are passing. Let me know if you don't agree with my last commit (01a504787).